### ISSUE TITLE
Rule Request: Prefer using Asset Symbols

### ISSUE THREAD

**Issue by 417-72KI** [realm/SwiftLint#5939]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New Rule Request

It would be happy when Image that initialized with string replaced with initializer with asset symbols.

> 1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

`UIKit.UIImage(named:)` and `SwiftUI.Image(_:)` contain the risk of bugs due to typos.
Since Xcode 15, Xcode generates codes for images in the Asset Catalog and it can avoid typos. ([ref](https://developer.apple.com/documentation/xcode-release-notes/xcode-15-release-notes#Asset-Catalogs))

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

### Would trigger:
```swift
// UIKit
UIImage(named: "some_image") // simple
UIImage(named: "some image")  // contains space
UIImage(named: "someImage", in: .module) // contains specified module

// SwiftUI
Image("some_image") // simple
Image("some image")  // contains space
Image("someImage", bundle: .module) // contains specified module
```

### Would not trigger:
```swift
// UIKit
UIImage(resource: .someImage)
UIImage(systemName: "trash")  // using SF Symbols

// SwiftUI
Image(.someImage)
Image(systemName: "trash")  // using SF Symbols
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

Currently no.

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Enabled by default on Swift 5.9+.
Asset symbol generation is enabled by default in Xcode, and it needs editing the build setting to disable it.

**Comment by danglingP0inter** [comment]
hey @SimplyDanny, I've raised a PR to add this rule. Please have a look.
